### PR TITLE
Remove an unused profile event code from object manager.

### PR DESCRIPTION
**Pull request by rkooo567**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/object_manager/object_manager.cc [hunk 1]
```diff
@@ -300,36 +300,9 @@ void ObjectManager::HandleSendFinished(const ObjectID &object_id, const NodeID &
                  << ", status: " << status.ToString();
   if (!status.ok()) {
     // TODO(rkn): What do we want to do if the send failed?
+    RAY_LOG(ERROR) << "Failed to send a push request for an object " << object_id
```
Comment (ericl) [inline src/ray/object_manager/object_manager.cc:303]: This will be too spammy--- please remove or set to DEBUG.

**Review by ericl — CHANGES_REQUESTED** [review]
LGTM but the error message is at the wrong level.

File: src/ray/object_manager/object_manager.cc (hunk 1 above)
Comment (rkooo567) [reply src/ray/object_manager/object_manager.cc:303]: Ah yeah good point... I blindly made it error for some reasons... Let me fix it.

**Review by ericl — APPROVED** [review]

**Comment by rkooo567** [comment]
TODO: Fix actor_broadcast test (seems like this feature is used by one test)

**Comment by rkooo567** [comment]
Hmm actually the test itself seems useful although there's only one test.

@ericl Do you think we can remove https://github.com/ray-project/ray/blob/6240d22060a011f9c248d6396bfaf50cb4a42d84/python/ray/tests/test_object_manager.py#L100? Otherwise, I will probably refactor profile code to a separate class and add an option to turn on/off

**Comment by rkooo567** [comment]
OR we can potentially replace the test to use events cc @scv119

**Comment by rkooo567** [comment]
If there's no objection, I will comment out the test, and implement the same test with the event framework in the follow up PR.

**Comment by scv119** [comment]
sounds good to me

**Review by scv119 — APPROVED** [review]

**Comment by rkooo567** [comment]
Windows tests + small & large test failures are flaky in the master.

**Pull request merged**